pkg/operation/seed/seed_test.go

```diff
@@ -144,4 +152,182 @@ var _ = Describe("seed", func() {
+					return fmt.Errorf("expect 200Gi found %v", *pvc.Spec.Resources.Requests.Storage())
+				}
+				return nil
+			}
+			objectOfTypePVC = gomock.AssignableToTypeOf(&corev1.PersistentVolumeClaim{})
+			objectOfTypeSTS = gomock.AssignableToTypeOf(&appsv1.StatefulSet{})
+		)
+
+		It("should patch garden/loki's PVC when new size is greater than the current one", func() {
+			runtimeClient.EXPECT().Get(ctx, lokiPVCKey, objectOfTypePVC).DoAndReturn(funcGetLokiPVC)
+			runtimeClient.EXPECT().Patch(ctx, statefulset, zeroReplicaRawPatch)
+			runtimeClient.EXPECT().Get(gomock.Any(), lokiStatefulSetKey, objectOfTypeSTS).DoAndReturn(funcGetScaledToZeroLokiStatefulset)
+			runtimeClient.EXPECT().Patch(ctx, objectOfTypePVC, gomock.AssignableToTypeOf(patch)).DoAndReturn(funcPatchTo200GiStorage)
+			runtimeClient.EXPECT().Delete(ctx, statefulset)
+			Expect(ResizeOrDeleteLokiDataVolumeIfStorageNotTheSame(ctx, runtimeClient, new200GiStorageQuantity, log)).To(Succeed())
+		})
+
+		It("should delete garden/loki's PVC when new size is less than the current one", func() {
+			runtimeClient.EXPECT().Get(ctx, lokiPVCKey, objectOfTypePVC).DoAndReturn(funcGetLokiPVC)
+			runtimeClient.EXPECT().Patch(ctx, statefulset, zeroReplicaRawPatch)
+			runtimeClient.EXPECT().Get(gomock.Any(), lokiStatefulSetKey, objectOfTypeSTS).DoAndReturn(funcGetScaledToZeroLokiStatefulset)
+			runtimeClient.EXPECT().Delete(ctx, lokiPVC)
+			runtimeClient.EXPECT().Delete(ctx, statefulset)
+			Expect(ResizeOrDeleteLokiDataVolumeIfStorageNotTheSame(ctx, runtimeClient, new80GiStorageQuantity, log)).To(Succeed())
+		})
+
+		It("shouldn't do anything when garden/loki's PVC is missing", func() {
+			runtimeClient.EXPECT().Get(ctx, lokiPVCKey, objectOfTypePVC).Return(errNotFound)
+			Expect(ResizeOrDeleteLokiDataVolumeIfStorageNotTheSame(ctx, runtimeClient, new80GiStorageQuantity, log)).To(Succeed())
+		})
+
+		It("shouldn't do anything when garden/loki's PVC storage is the same as the new one", func() {
+			runtimeClient.EXPECT().Get(ctx, lokiPVCKey, objectOfTypePVC).DoAndReturn(funcGetLokiPVC)
+			Expect(ResizeOrDeleteLokiDataVolumeIfStorageNotTheSame(ctx, runtimeClient, new100GiStorageQuantity, log)).To(Succeed())
+		})
+
+		It("should proceed with the garden/loki's PVC resizing when Loki sts is missing", func() {
+			runtimeClient.EXPECT().Get(ctx, lokiPVCKey, objectOfTypePVC).DoAndReturn(funcGetLokiPVC)
+			runtimeClient.EXPECT().Patch(ctx, statefulset, zeroReplicaRawPatch).Return(errNotFound)
+			runtimeClient.EXPECT().Patch(ctx, objectOfTypePVC, gomock.AssignableToTypeOf(patch)).DoAndReturn(funcPatchTo200GiStorage)
+			runtimeClient.EXPECT().Delete(ctx, statefulset).Return(errNotFound)
+			Expect(ResizeOrDeleteLokiDataVolumeIfStorageNotTheSame(ctx, runtimeClient, new200GiStorageQuantity, log)).To(Succeed())
+		})
+
+		It("should succeed with the garden/loki's PVC resizing when Loki sts was deleted during function execution", func() {
```

Comment:
Nit: for consistency with the other test names

```suggestion
		It("should succeed with the garden/loki's PVC resizing when Loki StatefulSet was deleted during function execution", func() {
```